File: usecases/blea-guest-serverless-api-sample/lambda/nodejs/listItems.js
'use strict';

const AWSXRay = require('aws-xray-sdk');
const AWS = AWSXRay.captureAWS(require('aws-sdk'));
const ddb = new AWS.DynamoDB();
const params = {
  TableName: process.env.DDB_TABLE,
};

module.exports.listItems = (event, context, callback) => {
  ddb.scan(params, (error, result) => {
    if (error) {
      console.error(error);
      const response = {
        statusCode: error.statusCode,
        body: error.message,
      };
      callback(null, response);
    } else {
      console.log('Success', result.Items);
      const response = {
        statusCode: 200,
        body: JSON.stringify(result.Items),
      };
      callback(null, response);
    }
  });
};

File: usecases/blea-guest-serverless-api-sample/lambda/python/listItems/listItems.py
import json
import os
import boto3
from typing import Any, Dict
from aws_lambda_powertools import Logger, Tracer
from aws_lambda_powertools.utilities.typing import LambdaContext
from aws_lambda_powertools.utilities.data_classes import APIGatewayProxyEvent
from botocore.exceptions import ClientError


logger = Logger()
tracer = Tracer()

dynamodb = boto3.resource("dynamodb")
table = dynamodb.Table(os.environ["DDB_TABLE"])

@logger.inject_lambda_context(log_event=True)
@tracer.capture_lambda_handler

def lambda_handler(event: APIGatewayProxyEvent, context: LambdaContext) -> Dict[str, Any]:
  logger.info('info')
  logger.debug('debug')

  try:
    response = table.scan()
    print(response)
  except ClientError as e:
    print(e.response['Error']['Message'])
  else:
    return {
      "statusCode": 200,
      "body" : json.dumps(response["Items"])
    }